safekeeper/src/send_interpreted_wal.rs

```diff
@@ -46,10 +63,29 @@ impl<IO: AsyncRead + AsyncWrite + Unpin> InterpretedWalSender<'_, IO> {
         keepalive_ticker.set_missed_tick_behavior(MissedTickBehavior::Skip);
         keepalive_ticker.reset();
 
+        let (tx, rx) = tokio::sync::mpsc::channel::<Batch>(2);
+        let batches_stream =
+            tokio_stream::wrappers::ReceiverStream::new(rx).then(|batch| async move {
```

Comment:
Inlined the decoding in the select branch in https://github.com/neondatabase/neon/pull/9821/commits/d0eaf20779349f037bab3e93a765c51750497b9e. It actually performs a bit better in benchmarks.

Reply:
Right, I had missed the concurrency aspect. I suspect it doesn't buy us much, since the reads will mostly be from the kernel page cache and the network send buffer will smooth out the network writes. Sounds like the benchmarks agreed.